tests/test_akafka.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 - 2022 Universität Tübingen, DKFZ and EMBL
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Testing Apache Kafka based providers."""
+
+import json
+import random
+import string
+
+from kafka import KafkaConsumer
+from testcontainers.kafka import KafkaContainer
+
+from hexkit.eventpub.providers.akafka import KafkaEventPublisher
+from hexkit.utils import exec_with_timeout
+
+
+def test_kafka_event_publisher():
+    """Test the KafkaEventPublisher."""
+    random_string = "".join(random.choices(string.ascii_letters, k=10))
+    event_payload = {"test_content": random_string}
+    event_type = "test_event"
+    event_key = "test_key"
+    topic_name = "test_topic"
+
+    with KafkaContainer() as kafka:
```

Comment:
We should also add unit tests for the KafkaEventPublisher that do not spin up a container and can test more edge cases more quickly. This integration test can then stay very simple. (Testing pyramid.)

Reply:
👍

Reply:
Done.